components/l10n/common/locale_util.cc

```diff
@@ -5,78 +5,94 @@
 
 #include "brave/components/l10n/common/locale_util.h"
 
-#include <algorithm>
-#include <vector>
-
-#include "base/strings/string_split.h"
-#include "base/strings/string_util.h"
-#include "base/strings/utf_string_conversions.h"
-#include "ui/base/resource/resource_bundle.h"
+#include "absl/types/optional.h"
+#include "base/no_destructor.h"
+#include "brave/components/l10n/browser/default_locale_util.h"
+#include "brave/components/l10n/common/locale_subtag_info.h"
+#include "brave/components/l10n/common/locale_subtag_parser_util.h"
 
 namespace brave_l10n {
 
 namespace {
 
-const char kDefaultLanguageCode[] = "en";
-const char kDefaultCountryCode[] = "US";
+constexpr char kFallbackLocale[] = "en_US";
+constexpr char kFallbackLanguageCode[] = "en";
+constexpr char kFallbackCountryCode[] = "US";
+
+absl::optional<std::string>& DefaultLocaleForTesting() {
+  static base::NoDestructor<absl::optional<std::string>> locale;
+  return *locale;
+}
+
+absl::optional<std::string>& LastDefaultLocaleForTesting() {
+  static base::NoDestructor<absl::optional<std::string>> last_locale;
+  return *last_locale;
+}
 
 }  // namespace
 
-std::string GetLanguageCode(
+ScopedDefaultLocaleForTesting::ScopedDefaultLocaleForTesting(
     const std::string& locale) {
-  const std::vector<std::string> locale_components = base::SplitString(locale,
-      ".", base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  LastDefaultLocaleForTesting() = DefaultLocaleForTesting();
```

Comment:
pls remove `LastDefaultLocaleForTesting()` and keep the last local in `ScopedDefaultLocaleForTesting` member.